libs/libc/stdio/lib_vprintf.c

```diff
@@ -49,7 +52,13 @@
 
 int vprintf(FAR const IPTR char *fmt, va_list ap)
 {
+#if CONFIG_NFILE_STREAMS > 0
   /* vfprintf into stdout */
 
   return vfprintf(stdout, fmt, ap);
+#else
+  /* vsyslog into debug channel */
+
+  return nx_vsyslog(LOG_INFO, fmt, &ap);
```

Comment:
```suggestion
  return vdprintf(1, fmt, &ap);
```
This is the perfect replacement!

Reply:
Done.